wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -245,6 +245,34 @@ public void stop(String workspaceId) throws NotFoundException, ServerException,
     }
 
     /**
+     * Adds machine into workspace runtime.
+     *
+     * @param workspaceId
+     *         id of workspace to which runtime machine should be added
+     * @param machine
+     *         machine to add to specified runtime
+     * @throws ConflictException
+     *         if given machine already exists in specified runtime
+     * @throws NotFoundException
+     *         if workspace with specified id not exists
+     */
+    public void addMachineIntoRuntime(String workspaceId, MachineImpl machine) throws ConflictException, NotFoundException {
```

Comment:
* I don't like the method name, `addMachine`(`injectMachine` maybe) is more than enough to understand what it does, for what reason `workspaceId` is used (p.s. see machine#getWorkspaceId)? 
* Locks used wrong(or event not used at all)
* The method doesn't check the workspace runtime state while it MUST do it, as what is the purpose of adding the machine to the starting workspace? 
* The method doesn't respect the current state of the `isPreDestroyInvoked` flag.